### PR TITLE
Add slash command '/test-gmt-dev' to test GMT dev version

### LINKED ISSUE #828: The workflow of "GMT Latest Tests" are triggered multiple times
**Description of the problem**

In my recent PR (#827), the workflow of "GMT Latest Tests" are triggered twice (https://github.com/GenericMappingTools/pygmt/actions/runs/524773762, and https://github.com/GenericMappingTools/pygmt/actions/runs/524773759), possibly because I asked two teams for review.

Can we improve the workflow triggering condition?
https://github.com/GenericMappingTools/pygmt/blob/8e2b3691f40b8e455885195f7f738bfc979dc845/.github/workflows/ci_tests_dev.yaml#L6-L13

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds the slash command `/test-gmt-dev` which can trigger the [GMT Dev Tests](https://github.com/GenericMappingTools/pygmt/blob/master/.github/workflows/ci_tests_dev.yaml) workflow. So we can test GMT master branch anytime, without requesting reviews from other maintainers or teams (Also fixes #828).

As you may know/remember, the slash command can only read the configurations in the master branch, so we can't test this PR unless we merge it into the master branch.

To test this PR, I forked the PyGMT repository to my own account (https://github.com/seisman/pygmt), and made changes directly to my own master branch.

If you [compare my master branch and the PyGMT master branch](https://github.com/GenericMappingTools/pygmt/compare/master...seisman:master), you will see my master branch is 2 commits ahead of PyGMT master branch. The two commits are:

- https://github.com/seisman/pygmt/commit/11446e2b13ae503f6eeb1a9c0a5bee43262434a7: This commit is the same as the only commit in this PR 
- https://github.com/seisman/pygmt/commit/37f83e4caaca2fb94f053b6f0ce2cca88814ae51: The slash command uses the GenericMappingTools bot to generate a token, which is not available in my own fork. So I have to use my own token instead. This commit is necessary in my own fork for testing purposes.

To confirm that everything works well, I opened an example PR in my own fork (https://github.com/seisman/pygmt/pull/2). Please go to that PR to see how it works.

**TODO:**

- [x] Remove `review_requested` condition from the workflow, so that we can run the GMT Latest Tests workflow by marking it "ready for review", or using the slash command `/test-gmt-dev` (done in 4a22c59aa2b3e04b4f018c13c1b5dc8d4e28dae9)
- [x] Update the [pull request template] (https://github.com/GenericMappingTools/pygmt/blob/master/.github/PULL_REQUEST_TEMPLATE.md) (done in 8620bab3fe1c00d9c0ff1e89b403b3a2a3e5f26b)
- [x] Update contributing guide or maintenance guide (done in ef31c150dabb74fa83f22aff7a0aae9dc2c40da6)

**Known issues:**

The "GMT Latest Tests" triggered by `/test-gmt-master` won't show the workflow status in the status summary page (like the screenshot below), so we have to go the the "Actions" page to find the testing results.
image

Note: you may find the following status page in PR https://github.com/seisman/pygmt/pull/2. These "GMT Latest Tests" status were triggered by "mark it ready for review", not by the slash command `/test-gmt-master`.
image

One possible solution is: when the workflow is triggered by slash commands, we can let the workflow leave a comment saying that "The workflow is running at https://github.com/GenericMappingTools/pygmt/actions/runs/XXXXX"

Fixes #828.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Notes**

- You can write `/format` in the first line of a comment to lint the code automatically

### CODE REVIEW

**Review by weiji14 — COMMENTED** [review]
Please also update the part in https://github.com/GenericMappingTools/pygmt/blob/v0.2.1/MAINTENANCE.md#github-actions which says:

>     ci_tests_dev.yaml (GMT Latest Tests on Linux/macOS/Windows).
> 
> This is only triggered when a review is requested or re-requested on a PR. It is also scheduled to run daily on the master branch.

to mention that the test can also be triggered using `/test-gmt-master`.


> Note: you may find the following status page in PR seisman#2. These "GMT Latest Tests" status were triggered by "mark it ready for review", not by the slash command /test-gmt-master.
image
> One possible solution is: when the workflow is triggered by slash commands, we can let the workflow leave a comment saying that "The workflow is running at https://github.com/GenericMappingTools/pygmt/actions/runs/XXXXX"

Might be complicated trying to code up the chatbot comment (though it will be nice, if not a bit annoying). How often will we need to keep triggering GMT Latest Tests? If it's not too often I think it's fine to just navigate to the Actions tab manually.

**Comment by seisman** [comment]
> Might be complicated trying to code up the chatbot comment (though it will be nice, if not a bit annoying). How often will we need to keep triggering GMT Latest Tests? If it's not too often I think it's fine to just navigate to the Actions tab manually.

We probably only want to trigger the GMT Latest Tests when a PR is close to merging.

File: .github/workflows/slash-command-dispatch.yml [hunk 1]
```diff
@@ -21,5 +21,6 @@ jobs:
           token: ${{ steps.generate-token.outputs.token }}
           commands: |
             format
+            test-gmt-master
```
Comment (weiji14) [inline .github/workflows/slash-command-dispatch.yml:24]:
```suggestion
            test-gmt-latest
```
What do you think about changing the name to 'latest' to match the Github Actions name?

Comment (seisman) [reply .github/workflows/slash-command-dispatch.yml:24]: When we say "the latest GMT", we usually mean "the latest stable GMT release" (i.e., 6.1.1). 

What about "GMT Dev Tests" and `/test-gmt-dev`?

Comment (weiji14) [reply .github/workflows/slash-command-dispatch.yml:24]: Yep, dev sounds good and is shorter too.

File: .github/workflows/ci_tests_dev.yaml [hunk 2]
```diff
@@ -7,7 +7,9 @@ on:
   # push:
   #   branches: [ master ]
   pull_request:
-    types: [review_requested, ready_for_review]
+    types: [ready_for_review]
+  repository_dispatch:
+    types: [test-gmt-master-command]
```
Comment (seisman) [inline .github/workflows/ci_tests_dev.yaml:12]:
```suggestion
    types: [test-gmt-dev-command]
```

File: MAINTENANCE.md [hunk 3]
```diff
@@ -62,8 +62,8 @@ On the *master* branch, the workflow also handles the documentation deployment:
 
 2. `ci_tests_dev.yaml` (GMT Latest Tests on Linux/macOS/Windows).
 
-This is only triggered when a review is requested or re-requested on a PR.
-It is also scheduled to run daily on the *master* branch.
+This is triggered when a PR is marked as "ready for review", or using the slash
+command `/test-gmt-master`. It is also scheduled to run daily on the *master* branch.
```
Comment (seisman) [inline MAINTENANCE.md:66]:
```suggestion
command `/test-gmt-dev`. It is also scheduled to run daily on the *master* branch.
```

File: .github/workflows/slash-command-dispatch.yml (hunk 1 above)
Comment (seisman) [reply .github/workflows/slash-command-dispatch.yml]: Done in 6df8e340cc9462efec6365ccdbe9f8c644da3fea and 72b7dfae6207e1461ae4a51295efbe1c2d1722ef

**Review by weiji14 — APPROVED** [review]

**Pull request merged**